tools/handshake-runner/Simulation.cpp

```diff
@@ -106,22 +106,53 @@ Any readValueWithType(mlir::Type type, std::string in) {
     arg >> x;
     APFloat aparg(x);
     return aparg;
+  } else if (auto tupleType = type.dyn_cast<TupleType>()) {
+    char tmp;
+    arg >> tmp;
+    assert(tmp == '(' && "tuple should start with '('");
+    std::vector<Any> values;
+    unsigned size = tupleType.getTypes().size();
+    values.reserve(size);
+    // Parse element by element
+    for (unsigned i = 0; i < size; ++i) {
+      values.push_back(readValueWithType(tupleType.getType(i), arg));
+      // Consumes either the ',' or the ')'
+      arg >> tmp;
+    }
+    assert(tmp == ')' && "tuple should end with ')'");
+    return values;
```

Comment:
I know that there's not much precedence in handshake-runner for non-assert verification of input arguments, but something to verify # of provided args vs. tuple size would be nice. However, it's probably better to keep this for a follow-up commit to fix them all at once.

Reply:
Currently, this is all a bit fragile, I agree. 
As you suggest, while changing that, will improve this code, it will also touch multiple things not related to this change.